ochap-backend/src/main/java/ch/heigvd/amt/backend/routes/ShoppingCartTemplate.java

```diff
@@ -0,0 +1,113 @@
+package ch.heigvd.amt.backend.routes;
+
+import ch.heigvd.amt.backend.entities.ProductQuantity;
+import ch.heigvd.amt.backend.entities.ShoppingCart;
+import ch.heigvd.amt.backend.repository.ProductQuantityDAO;
+import ch.heigvd.amt.backend.repository.ShoppingCartDAO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+
+import javax.persistence.EntityManager;
+import javax.validation.Valid;
+import java.util.List;
+import java.util.Optional;
+
+@Controller
+public class ShoppingCartTemplate {
+
+  @Autowired
+  private ShoppingCartDAO shoppingCartDAO;
+  @Autowired
+  private ProductQuantityDAO productQuantityDAO;
+  @Autowired
+  private EntityManager em;
+
+  @GetMapping("/shopping-cart")
+  public String basicTemplate(Model model) {
+    int clientId = 2;
+    List<ProductQuantity> shoppingCart = getProductQuantitesByClientId(clientId).getBody();
+    ShoppingCart cart = getShoppingCartByClientId(clientId).getBody();
+
+    ProductQuantity[] items;
+    if (shoppingCart != null) {
+      items = shoppingCart.toArray(ProductQuantity[]::new);
+    } else {
+      items = new ProductQuantity[] {};
+    }
+
+    model.addAttribute("products", items);
+    model.addAttribute("postAddress", "/shopping-cart");
+    model.addAttribute("cart", cart);
+    return "shopping-cart-template";
```

Comment:
```suggestion
    return "shopping-cart";
```

No point on adding the `template` suffix, it's already in the 'templates' directory.